plugins/filter/quote.py

```diff
@@ -46,21 +52,22 @@ def _quote_cmd(s):
     # 'file &whoami.exe' would result in 'whoami.exe' being executed and then that output being used as the argument
     # instead of the literal string.
     # https://stackoverflow.com/questions/3411771/multiple-character-replace-with-python
-    for c in '^()%!"<>&|':  # '^' must be the first char that we scan and replace
+    for c in u'^()%!"<>&|':  # '^' must be the first char that we scan and replace
         if c in s:
-            s = s.replace(c, "^" + c)
+            # I can't find any docs that explicitly say this but to escape ", it needs to be prefixed with \^.
+            s = s.replace(c, (u"\\^" if c == u'"' else u"^") + c)
 
-    return '^"{0}^"'.format(s)
+    return u'^"{0}^"'.format(s)
 
 
 def _quote_pwsh(s):
     # https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_quoting_rules?view=powershell-5.1
     if not s:
-        return "''"
+        return u"''"
 
     # We should always quote values in PowerShell as it has conflicting rules where strings can and can't be quoted.
-    # This means we quote the entire arg with single quotes and escape the single quotes already in the arg.
-    return "'{0}'".format(s.replace("'", "''"))
+    # This means we quote the entire arg with single quotes and just double up on the single quote equivalent chars.
+    return u"'{0}'".format(_UNSAFE_PWSH.sub(u'\\1\\1', s))
```

Comment:
Referring to my comment above, the character class would need single quote `'` so the sub catches it no?